### PR TITLE
Project refactor

### PR DESCRIPTION
**Pull request by javrudsky**
- Moved test to split framework project
- Removed CocoaPods dependency
- Linked test framework through carthage to Split framework project
- Deprecated config methods in favor of properties to stick to Uniform Access Principle
- Several minor refactors within the project
- Replaced Http Stub library MockingJay by OHHTTPStubs because of lack of support for Carthage.

### CODE REVIEW

File: Split.podspec [hunk 1]
```diff
@@ -18,4 +18,10 @@ This SDK is designed to work with Split, the platform for controlled rollouts, s
   s.source_files = 'Split/*.{swift}'
   s.source_files = 'Split/**/*.{swift}'
 
+  s.test_spec 'SplitTests' do |t|
```
Comment (SenhorCastor) [inline Split.podspec]: review this

File: Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift [hunk 2]
```diff
@@ -22,7 +22,7 @@ import Foundation
         self.changeNumber = changeNumber
     }
     
-    public func addSegments(segmentNames: [String]) {
+    public func addSegments(segmentNames: [String], key: String = "") {
```
Comment (SenhorCastor) [inline Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift]: can you review this?

File: Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift [hunk 3]
```diff
@@ -34,8 +34,12 @@ import Foundation
         self.mySegments.removeAllObjects()
     }
     
-    public func getSegments() -> [String] {
-        return self.mySegments.allKeys as! [String]
+    public func getSegments() -> [String]? {
+        return self.mySegments.allKeys as? [String]
+    }
+    
+    public func getSegments(key: String) -> [String]? {
```
Comment (SenhorCastor) [inline Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift]: can you check this?

Comment (SenhorCastor) [reply Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift]: check if we can remove this call that does not use key at all and behaves the same as getSegments with no key

File: Split/HttpSplitChangeFetcher.swift [hunk 4]
```diff
@@ -48,7 +48,7 @@ public enum FecthingPolicy {
             }
             semaphore.wait()
             
-            guard let change: SplitChange = try requestResult!.unwrap() else {
+            guard let change: SplitChange = try requestResult!.unwrap(), change.isValid else {
```
Comment (SenhorCastor) [inline Split/HttpSplitChangeFetcher.swift]: Can you check that this error is logged but does not cause this to crash?

File: Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift (hunk 3 above)
Comment (javrudsky) [reply Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift]: This is related to the previous comment about InMemoryMySegmentsCache$.addSegments methods.

File: Split/HttpSplitChangeFetcher.swift (hunk 4 above)
Comment (javrudsky) [reply Split/HttpSplitChangeFetcher.swift]: This function is called from RefreshableSplitFetcher cache and it's the error is handled correctly . No crash should happen here.

File: Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift (hunk 2 above)
Comment (SenhorCastor) [reply Split/Fetcher Engine/Cache/InMemoryMySegmentsCache.swift]: Ok, I will create a ticket and we will prioritize

**Pull request merged**